flask_sqlalchemy/__init__.py

```diff
@@ -874,8 +874,13 @@ def apply_driver_hacks(self, app, sa_url, options):
         if sa_url.drivername.startswith('mysql'):
             sa_url.query.setdefault('charset', 'utf8')
             if sa_url.drivername != 'mysql+gaerdbms':
-                options.setdefault('pool_size', 10)
-                options.setdefault('pool_recycle', 7200)
+                pool_class_name = None
+                if 'poolclass' in options:
+                    pool_class_name = pool_class.__name__
```

Comment:
Where is `pool_class` coming from? It was being loaded from the dict.

Reply:
Yes good spot - fixed